tests/framework/extensions/prime/primechecks.go

```diff
@@ -0,0 +1,92 @@
+package primechecks
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	client "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/rancherversion"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	PodResourceSteveType = "pod"
+)
+
+// CheckUIBrand checks the UI brand of Rancher Prime. If the Rancher instance is not Rancher Prime, the UI brand should be blank.
+func CheckUIBrand(client *rancher.Client, isPrime bool, brand *client.Setting, rancherBrand string) (string, error) {
+	if isPrime {
+		if rancherBrand == brand.Value {
+			logrus.Infof("Rancher Prime UI brand: %s", brand.Value)
+			return brand.Value, nil
+		}
+	}
+
+	return brand.Value, nil
+}
+
+// CheckVersion checks the if Rancher Prime is set to true and the version of Rancher.
+func CheckVersion(isPrime bool, rancherVersion string, serverConfig *rancherversion.Config) (string, error) {
+	if isPrime {
+		if rancherVersion == serverConfig.RancherVersion {
+			logrus.Infof("Rancher Prime: %t | Version: %s", isPrime, serverConfig.RancherVersion)
+			return serverConfig.RancherVersion, nil
+		}
+	}
+
+	return serverConfig.RancherVersion, nil
+}
+
+// CheckSystemDefaultRegistry checks if the system default registry is set to the expected value.
+func CheckSystemDefaultRegistry(isPrime bool, primeRegistry string, registry *client.Setting) (string, error) {
+	if isPrime {
+		if primeRegistry == registry.Value {
+			logrus.Infof("Rancher Prime system default registry: %s", registry.Value)
+			return registry.Value, nil
+		}
+	}
+
+	return registry.Value, nil
+}
+
+// CheckLocalClusterRancherImages checks if the Rancher images are set to the expected registry.
+func CheckLocalClusterRancherImages(client *rancher.Client, isPrime bool, rancherVersion, primeRegistry, clusterID string) ([]string, []error) {
+	downstreamClient, err := client.Steve.ProxyDownstream(clusterID)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	steveClient := downstreamClient.SteveType(PodResourceSteveType)
+
+	pods, err := steveClient.List(nil)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	var imageResults []string
+	var imageErrors []error
+
+	for _, pod := range pods.Data {
+		podStatus := &corev1.PodStatus{}
+		err = v1.ConvertToK8sType(pod.Status, podStatus)
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		image := podStatus.ContainerStatuses[0].Image
+		image = image[:strings.IndexByte(image, '/')]
```

Comment:
an image using a registry for rancher images _should_ have more than 2 indices. 
`<registry>/<docker-repo>/<image-name>:<version>`

however for a non-registry setup, the image check is using the <docker-repo> slice instead of the registry (because there isn't a registry). . 

This can be a new item if you think it is outside the scope of this ticket, but we should have this pass for a non-prime setup when the registry isn't set.  

you could special case the below if/else statement `if image == "rancher"` and `!isPrime` then still count the image result as valid instead of error

or you could check the length of `strings.IndexByte(image, '/')`, if the length is 3, check for prime registry, otherwise, the test could just pass without validating the registry of the images?

Reply:
Resolved.